lib/api/objectGetLegalHold.js

```diff
@@ -0,0 +1,104 @@
+const async = require('async');
+const { errors, s3middleware } = require('arsenal');
+
+const { decodeVersionId, getVersionIdResHeader }
+    = require('./apiUtils/object/versioning');
+
+const { metadataValidateBucketAndObj } = require('../metadata/metadataUtils');
+const { pushMetric } = require('../utapi/utilities');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+
+const { convertToXml } = s3middleware.objectLegalHold;
+
+/**
+ * Returns legal hold status of object
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+function objectGetLegalHold(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'objectGetLegalHold' });
+
+    const { bucketName, objectKey, query } = request;
+
+    const decodedVidResult = decodeVersionId(query);
+    if (decodedVidResult instanceof Error) {
+        log.trace('invalid versionId query', {
+            versionId: query.versionId,
+            error: decodedVidResult,
+        });
+        return process.nextTick(() => callback(decodedVidResult));
+    }
+    const versionId = decodedVidResult;
+
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        objectKey,
+        requestType: 'objectGetLegalHold',
+        versionId,
+    };
+
+    return async.waterfall([
```

Comment:
no need to return here, async.waterfall will return undefined every time anyway ( from the docs anyway )